src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -411,44 +430,87 @@ public void makeStopToPointDistanceTables (Geometry treeRebuildZone) {
                 }
-                return stopToPointDistanceTables.get(stopIndex);
+                return stopToPointLinkageCostTables.get(stopIndex);
+            }
+
+            int[] linkageCostToPoints;
+
+            if (streetMode == StreetMode.WALK) {
+                // Walking distances from stops to street vertices are saved in the transitLayer.
+                // Get the pre-computed distance table from the stop to the street vertices,
+                // then extend that table out from the street vertices to the points in this PointSet.
+                TIntIntMap distanceTableToVertices = transitLayer.stopToVertexDistanceTables.get(stopIndex); // walk!
+                Envelope distanceTableZone = stopPoint.getEnvelopeInternal();
+                GeometryUtils.expandEnvelopeFixed(distanceTableZone, TransitLayer.WALK_DISTANCE_LIMIT_METERS);
+                linkageCostToPoints = distanceTableToVertices == null ? null :
+                        extendDistanceTableToPoints(distanceTableToVertices, distanceTableZone);
+
+            } else if (streetMode == StreetMode.BICYCLE) {
+                // Biking distances from stops to street vertices are not saved in the transitLayer, so additional
+                // steps are needed compared to Walk.
+                StreetRouter sr = new StreetRouter(transitLayer.parentNetwork.streetLayer);
+                sr.streetMode = StreetMode.BICYCLE;
+                sr.distanceLimitMeters = BICYCLE_DISTANCE_LINKING_LIMIT_METERS;
+                sr.quantityToMinimize = linkageCostUnit;
+                sr.setOrigin(transitLayer.streetVertexForStop.get(stopIndex));
+                sr.route();
+                Envelope distanceTableZone = stopPoint.getEnvelopeInternal();
+                GeometryUtils.expandEnvelopeFixed(distanceTableZone, BICYCLE_DISTANCE_LINKING_LIMIT_METERS);
+                linkageCostToPoints = extendDistanceTableToPoints(sr.getReachedVertices(), distanceTableZone);
+
+            } else if (streetMode == StreetMode.CAR) {
+                // The speeds for Walk and Bicycle can be specified in an analysis request, so it makes sense above to
+                // store distances and apply the requested speed. In contrast, car speeds vary by link and cannot be
+                // set in analysis requests, so it makes sense to use seconds directly as the linkage cost.
+                linkageCostUnit = StreetRouter.State.RoutingVariable.DURATION_SECONDS;
+                // TODO confirm this works as expected when modifications can affect street layer.
+                StreetRouter sr = new StreetRouter(transitLayer.parentNetwork.streetLayer);
+                sr.streetMode = StreetMode.CAR;
+                sr.timeLimitSeconds = CAR_TIME_LINKING_LIMIT_SECONDS;
+                sr.quantityToMinimize = linkageCostUnit;
+                sr.setOrigin(stopPoint.getY() / FIXED_FACTOR, stopPoint.getX() / FIXED_FACTOR);
```

Comment:
Should this setOrigin call be the same as the one in the above `StreetMode.BICYCLE` clause? It seems to just use the index number of the street vertex corresponding to the stop.

Reply:
Yes, forgot to commit this line.